highfive/newpr.py

```diff
@@ -230,27 +230,46 @@ def is_new_contributor(self, username, owner, repo):
             else:
                 raise e
 
-    def find_reviewer(self, msg):
+    def get_groups(self):
+        groups = deepcopy(self.repo_config['groups'])
+
+        # fill in the default groups, ensuring that overwriting is an
+        # error.
+        global_ = self._load_json_file('_global.json')
+        for name, people in global_['groups'].items():
+            assert name not in groups, "group %s overlaps with _global.json" % name
+            groups[name] = people
+
+        return groups
+
+    def find_reviewer(self, msg, exclude):
         """
         If the user specified a reviewer, return the username, otherwise returns
         None.
         """
         if msg is not None:
             match = reviewer_re.search(msg)
-            return match.group(1) if match else None
+            if match:
+                if match.group(1):
+                    # assign someone from the specified team
+                    groups = self.get_groups()
+                    potential = groups.get(match.group(2))
+                    if potential is None:
+                        potential = groups.get("%s/%s" % (match.group(1), match.group(2)))
+                    if potential is None:
+                        potential = groups["all"]
+                    else:
+                        potential.extend(groups["all"])
```

Comment:
Does this mean `r? @rust-lang/compiler` will potentially pick someone outside that group? (I'm trying to figure out why we want this branch)

Reply:
I *think* I intended this for repos outside of `rust-lang/rust`, e.g. embedded folks. In `rust-lang/rust` the `all` team.

Reply:
Oh I see 

https://github.com/rust-lang/highfive/blob/ab9ad47a683b488de3d85c69445c8daced06c8df/highfive/configs/rust-lang/rust.json#L3